internal/rpc/rpcserver/server.go

```diff
@@ -2437,6 +2445,10 @@ func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr
 	var vspHost string
 	var vspPubKey string
 	if req.VspHost != "" || req.VspPubkey != "" {
+		if req.VotingAddress != "" || req.PoolAddress != "" {
+			return status.Errorf(codes.InvalidArgument,
+				"request contains both legacy stakepoold and vspd options.")
+		}
```

Comment:
if you keep the check above, this one can be removed (we've already determined that both are not set together)